### ISSUE TITLE
MD5 validation does not work with blob encrypted with client-side mechanism

### ISSUE THREAD

**Issue by thchuong-x2011** [Azure/azure-storage-java#124]
By default, downloading blob from Azure storage will verify the MD5 hash of the blob content.
This step is done by comparing the MD5 in the metadata of the blob with that of the downloaded content. 
In the case of an encrypted blob, this means comparing the MD5 of the encrypted blob with that of the decrypted blob.

Attachment: a piece of code to reproduce the problem
[TestClientSideEncryption.java.txt](https://github.com/Azure/azure-storage-java/files/487731/TestClientSideEncryption.java.txt)

**Comment by pemari-msft** [comment]
\cc @asorrin-msft

**Comment by asorrin-msft** [comment]
Good catch, and thanks for the detailed repro!  Turns out there's a bug in the openInputStream() API, as well as a bug in our tests that disables the MD5 validation in some test scenarios.
I'll be working on a fix.  In the meantime, I think that the downloadToStream() API is unaffected by this issue, only openInputStream() (still need to fully validate this).

**Comment by chrisN777** [comment]
Hi @asorrin-msft, any progress on this issue? 
BTW: The workaround via download() is possible and not affected by this bug.
